dashboard/modules/job/job_manager.py

```diff
@@ -538,6 +542,36 @@ def _get_supervisor_runtime_env(
         runtime_env["env_vars"] = env_vars
         return runtime_env
 
+    async def _get_scheduling_strategy(self) -> SchedulingStrategyT:
+        if os.environ.get(RAY_JOB_ALLOW_DRIVER_ON_WORKER_NODES_ENV_VAR, "0") == "1":
+            logger.info(
+                f"{RAY_JOB_ALLOW_DRIVER_ON_WORKER_NODES_ENV_VAR} was set to 1. "
+                "Using Ray's default actor scheduling strategy for the job "
+                "driver instead of running it on the head node."
+            )
+            scheduling_strategy = "DEFAULT"
+        else:
+            head_node_id = (
+                await self._gcs_aio_client.internal_kv_get(
+                    "head_node_id".encode(),
+                    namespace=ray_constants.KV_NAMESPACE_JOB,
+                    timeout=30,
+                )
+            ).decode()
+            if head_node_id is None:
```

Comment:
this won't happen because `internal_kv_get` returns `Optional[bytes]` and then you already called `decode()` on it (which might error), you might want to defer the `decode()` until after this check.